hub/core/lock.py

```diff
@@ -108,15 +146,31 @@ def release(self):
             terminate_thread(self._thread)
             self._acquired = False
         try:
-            del self.storage[get_dataset_lock_key()]
+            del self.storage[self.path]
         except Exception:
             pass
 
 
 _LOCKS: Dict[str, Lock] = {}
 
 
-def lock(storage: StorageProvider, callback: Optional[Callable] = None):
+def _get_lock_key(storage_path: str, version: Optional[str] = None):
+    if version is None:
+        version = FIRST_COMMIT_ID
+    return storage_path + ":" + version
+
+
+def _get_lock_file_path(version: Optional[str] = None):
+    if version in (None, FIRST_COMMIT_ID):
+        return get_dataset_lock_key()
+    return "versions/" + version + "/" + get_dataset_lock_key()
+
+
+def lock_version(
```

Comment:
I have strong sense that this have to be action of the `StorageProvider`

Reply:
Hmm, the storage provider doesn't know anything about versions though.
Or are you saying that the storage provider should have a method to acquire a lock on any file in general?

Reply:
@AbhinavTuli You are right, I haven't thought of it. Perhaps  we luck of  abstraction `Version` to be able perform lock operation in on.  

I don't think we should lock any file. What might be the use case of it ? Seems like all other files are versioned and each individual works with their own version.

Let's keep it as is, than.